src/project.coffee

```diff
@@ -39,6 +39,14 @@ class Project extends Model
     @emitter = new Emitter
     @buffers ?= []
 
+    @directoryProviders = [new DefaultDirectoryProvider()]
+    atom.packages.serviceHub.consume(
+      'atom.directory-provider',
+      '^0.1.0',
+      # New providers are added to the front of @directoryProviders because
+      # DefaultDirectoryProvider is a catch-all that will always provide a Directory.
+      (provider) => @directoryProviders.splice(0, 0, provider))
```

Comment:
:+1: @maxbrunsfeld I agree with your thoughts on this. It also ties into @bolinfest's idea about having package metadata for the service provider prefix. Another wild thought... could we use the provider's service name make the protocol explicit? `atom.directory-provider.webdav`? Not sure it's a good idea.